File: Leetcode/3010. Divide an Array Into Subarrays With Minimum Cost I/solution.py
class Solution:
    def minimumCost(self, nums: List[int]) -> int:
        min1, min2, min3 = nums[0], nums[1], nums[2]

        for i in range(2, len(nums)):
            if min2 > nums[i]:
                min3 = min2
                min2 = nums[i]
            elif min3 > nums[i]:
                min3 = nums[i]
        
        return min1 + min2 + min3

File: Leetcode/3010. Divide an Array Into Subarrays With Minimum Cost I/solution.go
func minimumCost(nums []int) int {
	min1 := nums[0]
	min2 := nums[1]
	min3 := nums[2]

	for i := 2; i < len(nums); i++ {
		if min2 > nums[i] {
			min3 = min2
			min2 = nums[i]
		} else if min3 > nums[i] {
			min3 = nums[i]
		}
	}

	return min1 + min2 + min3
}